Tests/SwiftFusionTests/Core/VectorTests.swift

```diff
@@ -1,292 +1,192 @@
-    }
-
-    let v1 = makeVector(from: 1, stride: 1)
-    let v2 = makeVector(from: 2, stride: 1)
-    XCTAssertEqual(doMutatingAdd(v1, v2), makeVector(from: 3, stride: 2))
-
-    let (value, pb) = valueWithPullback(at: v1, v2, in: doMutatingAdd)
-    XCTAssertEqual(value, makeVector(from: 3, stride: 2))
-    for v in basisVectors {
-      XCTAssertEqual(pb(v).0, v)
-      XCTAssertEqual(pb(v).1, v)
-    }
-  }
-
-  /// Tests the value and derivative of the functional +.
-  func testAddFunctional() {
-    let v1 = makeVector(from: 1, stride: 1)
-    let v2 = makeVector(from: 2, stride: 1)
-    XCTAssertEqual(v1 + v2, makeVector(from: 3, stride: 2))
-
-    let (value, pb) = valueWithPullback(at: v1, v2, in: +)
-    XCTAssertEqual(value, makeVector(from: 3, stride: 2))
-    for v in basisVectors {
-      XCTAssertEqual(pb(v).0, v)
-      XCTAssertEqual(pb(v).1, v)
-    }
-  }
-
-  /// Tests the value and derivative of the mutating -=.
-  func testSubtractMutating() {
-    func doMutatingSubtract(_ v1: Testee, _ v2: Testee) -> Testee {
-      var result = v1
-      result -= v2
-      return result
-    }
-
-    let v1 = makeVector(from: 1, stride: 1)
-    let v2 = makeVector(from: 1, stride: -1)
-    XCTAssertEqual(doMutatingSubtract(v1, v2), makeVector(from: 0, stride: 2))
-
-    let (value, pb) = valueWithPullback(at: v1, v2, in: doMutatingSubtract)
-    XCTAssertEqual(value, makeVector(from: 0, stride: 2))
-    for v in basisVectors {
-      XCTAssertEqual(pb(v).0, v)
-      XCTAssertEqual(pb(v).1, -v)
+extension Vector {
+  /// XCTests `self`'s semantic conformance to `Vector`, expecting its scalars to match
+  /// `expectedScalars`.
+  ///
+  /// - Requires: `!distinctScalars.elementsEqual(self.scalars)`.
+  /// - Complexity: O(N²), where N is `self.dimension`.
+  func checkVectorSemantics<S1: Collection, S2: Collection>(
+    expecting expectedScalars: S1,
+    writing distinctScalars: S2,
+    maxSupportedScalarCount: Int = Int.max
+  ) where S1.Element == Double, S2.Element == Double {
+
+    // MARK: - Check the `scalars` property semantics.
+
+    self.scalars.checkCollectionSemantics(
+      expecting: expectedScalars, maxSupportedCount: maxSupportedScalarCount)
+
+    var mutableScalars = self.scalars
+    mutableScalars.checkMutableCollectionSemantics(writing: distinctScalars)
+
+    // Check that setting `scalars` actually changes it.
+    var mutableSelf = self
+    for (i, e) in zip(mutableSelf.scalars.indices, distinctScalars) { mutableSelf.scalars[i] = e }
+    XCTAssertTrue(mutableSelf.scalars.elementsEqual(distinctScalars))
+
+    // MARK: - Check the `dimension` property semantics.
+
+    XCTAssertEqual(self.dimension, expectedScalars.count)
+
+    // MARK: - Check the mathematical vector operation (`+`, `+=`, `-`, `-=`, `*`, `*=`, `dot`,
+    // `zeroValue`) semantics.
+
+    /// Returns a vector with the same `scalars.indices` as `self` but with the scalars replaced
+    /// by the stride from `start` by `stride`.
```

Comment:
I can't parse this doc.  The two “by”s are confusing.  Please clarify.

Reply:
Done